mitxgraders/sampling.py

```diff
@@ -338,6 +349,83 @@ class RealMatrices(RealMathArrays):
     })
 
 
+class AbstractSquareMatrices(VariableSamplingSet):  # pylint: disable=abstract-method
+    """
+    Abstract class representing a collection of square matrices
+
+    Config:
+    =======
+        dimension: Positive integer that specifies the dimension of the matrix (default 2)
+    """
+
+    # This is an abstract base class
+    __metaclass__ = abc.ABCMeta
+
+    # Store the dimension of the square matrix
+    schema_config = Schema({
+        Required('dimension', default=2): Positive(int)
+    })
+
+
+class IdentityMatrixMultiples(AbstractSquareMatrices):
+    """
+    Class representing a collection of multiples of the identity matrix
+    of a given dimension
+
+    Config:
+    =======
+        dimension: Positive integer that specifies the dimension of the matrix (default 2)
+        sampler: A scalar sampling set for the multiplicative constant (default [1, 5])
+
+    Usage:
+    ======
+
+    By default, we generate 2x2 matrices:
+    >>> matrices = IdentityMatrixMultiples()
+    >>> matrices.gen_sample().shape
+    (2, 2)
+
+    We can generate NxN matrices by specifying the dimension:
+    >>> matrices = IdentityMatrixMultiples(dimension=4)
+    >>> matrices.gen_sample().shape
+    (4, 4)
+
+    The scalar multiple can be generated in a number of ways:
+    >>> matrices = IdentityMatrixMultiples(sampler=[1,3])
+    >>> matrices = IdentityMatrixMultiples(sampler=(1, 3, 5))
```

Comment:
I'm going to remove this line and merge—this looks like old test before you removed DiscreteSet as asampler

Reply:
Ah, my bad. Thanks!